tensorboard/components_polymer3/tf_paginated_view/tf-dom-repeat.ts

```diff
@@ -0,0 +1,241 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {LegacyElementMixin} from '@polymer/polymer/lib/legacy/legacy-element-mixin';
+import {
+  PolymerDeepPropertyChange,
+  PolymerSpliceChange,
+} from '@polymer/polymer/interfaces';
+import {
+  templatize,
+  TemplateInstanceBase,
+} from '@polymer/polymer/lib/utils/templatize';
+import {property, observe} from '@polymer/decorators';
+
+import {ArrayUpdateHelper} from '../tf_dashboard_common/array-update-helper';
+
+/**
+ * The TfDomRepeatBehavior re-implements part of the dom-repeat template.
+ * It can be applied to any Polymer components to make it dom-repeat-like.
+ * The major discrepency is in syntax when using:
+ *
+ *   <template is="dom-repeat">
+ *     ...
+ *   </template>
+ *   // versus
+ *   <foo-comp>
+ *     <template>
+ *       ...
+ *     </template>
+ *   </foo-comp>
+ *
+ * When implementing the behavior, the Polymer component has to invoke
+ * `this.updateDom(newItems)`. Other protected APIs are `setGetItemKey` and
+ * `setCacheSize` (please see respective doc for details).
+ */
+export class TfDomRepeat<T extends {}> extends LegacyElementMixin(
+  ArrayUpdateHelper
```

Comment:
Hmm, interesting; I wonder how we would express a component that wanted
both the `ArrayUpdateHelper` and `DataLoaderBehavior` mixins?

Reply:
I think you do `ArrayUpdateHelper(DataLoaderBehavior(Polymer...)` or use `mixinBehaviors` property. Not sure which one is better yet.